File: abc174/4.py
n = input()
list_colors = [int(x) for x in input().split()]

# WRの順番は許容されない -> RW, RR, WW
# 1個の色を変更 or 2個の入れ替え -> どういった条件でどちらを選択するか?

i, k = 0, 1
while True:
    if list_colors[i] == "W" and list_colors[k] == "R":
        list_colors[i], list_colors[k] == "R", "W"
    if list_colors[i] == "W": # WW
        k += 1
    if list_colors[k] == "R": #RR
        i += 1
     

File: ts/abc234/4.ts
import * as fs from 'fs';

const input: any = fs.readFileSync('/dev/stdin', 'utf8').split(/\n/);

// output();
const output = (x: any) => console.log(x)

const max_num: number = input[0].split(" ")[0]
const k: number = input[0].split(" ")[1]

const value_array: [number] = input[1].split(' ').map((x: string) => Number(x));

for (let index = k; index <= max_num; index++) {
	let temp_array: number[] = value_array.slice(0, index).reverse()
	output(temp_array[k-1])
}

